include/alpaka/core/ApiCudaRt.hpp

```diff
@@ -0,0 +1,357 @@
+#pragma once
+
+#ifdef ALPAKA_ACC_GPU_CUDA_ENABLED
+
+#    include <alpaka/core/BoostPredef.hpp>
+
+#    include <cuda_runtime_api.h>
+
+namespace alpaka
+{
+    struct ApiCudaRt
+    {
+        // names
+        static constexpr char name[] = "Cuda";
+        static constexpr auto version = BOOST_LANG_CUDA;
+
+        // types
+        using DeviceProp_t = ::cudaDeviceProp;
+        using Error_t = ::cudaError_t;
+        using Event_t = ::cudaEvent_t;
+        using Stream_t = ::cudaStream_t;
+        using Pos_t = ::cudaPos;
+        using Extent_t = ::cudaExtent;
+        using FuncAttributes_t = ::cudaFuncAttributes;
+        using MemcpyKind_t = ::cudaMemcpyKind;
+        using Memcpy3DParms_t = ::cudaMemcpy3DParms;
+        using PitchedPtr_t = ::cudaPitchedPtr;
+
+        // constants
+        static constexpr auto success = ::cudaSuccess;
```

Comment:
```suggestion
        inline static constexpr auto success = ::cudaSuccess;
```
This only makes a difference if the constant is ODR-used anywhere (e.g. its address is taken). I would add it though in case someone would like to do that. Applies to all constants in this struct and in `ApiHipRt`.

Reply:
According to [cppreference.com](https://en.cppreference.com/w/cpp/language/constexpr)
> A `constexpr` specifier used in a function or `static` data member (since C++17) declaration implies `inline`.

so the explicit `inline` should not be necessary.

I can still add it if you prefer, though.

Reply:
Cool! I learned sth! You can omit the `inline` then.